lib/kitchen/driver/ec2.rb

```diff
@@ -786,7 +786,10 @@ def create_security_group(state)
 
                    subnets.first.vpc_id
                  elsif config[:subnet_filter]
-                   subnets = ec2.client.describe_subnets(filters: [{ name: "tag:#{config[:subnet_filter][:tag]}", values: [config[:subnet_filter][:value]] }]).subnets
+                   filters = config[:subnet_filter].fetch(:filters, {}).map do |key, value|
+                     {name: key, values: [value]}
+                   end
+                   subnets = ec2.client.describe_subnets(filters: filters).subnets
                    raise "Subnets with tag '#{config[:subnet_filter][:tag]}=#{config[:subnet_filter][:value]}' not found during security group creation" if subnets.empty?
```

Comment:
I think we need to update the error message.

Reply:
It does that now:

 > Failed to complete #create action: [Subnets with filters: '[{:name=>:mapPublicIpOnLaunch, :values=>["true"]}, {:name=>:"tag:Name", :values=>["foo"]}]' not found during security group creation] on test-ec2